src/daemon/plugin.h

```diff
@@ -42,17 +42,26 @@
 # define DATA_MAX_NAME_LEN 128
 #endif
 
-#define DS_TYPE_COUNTER  0
-#define DS_TYPE_GAUGE    1
-#define DS_TYPE_DERIVE   2
-#define DS_TYPE_ABSOLUTE 3
+#define DS_TYPE_COUNTER   0
+#define DS_TYPE_GAUGE     1
+#define DS_TYPE_DERIVE    2
+#define DS_TYPE_ABSOLUTE  3
+#define DS_TYPE_GAUGE_INT 4
```

Comment:
We don't want to be changing the basic `collectd` structures -- this is about the type as it's being sent into the `createTimeSeries` endpoint, so I would instead just add a metadata key to override the value type (`double`->`int`) while keeping the value kind (`gauge` vs. `cumulative`).